prefect_azure/container_instance.py

```diff
@@ -0,0 +1,704 @@
+    cpu: float = Field(
+        title="CPU",
+        default=ACI_DEFAULT_CPU,
+        description=(
+            "The number of virtual CPUs to assign to the task container. "
+            f"If not provided, a default value of {ACI_DEFAULT_CPU} will be used."
+        ),
+    )
+    gpu_count: Optional[int] = Field(
+        title="GPU Count",
+        default=None,
+        description=(
+            "The number of GPUs to assign to the task container. "
+            "If not provided, no GPU will be used."
+        ),
+    )
+    gpu_sku: Optional[str] = Field(
+        title="GPU SKU",
+        default=None,
+        description=(
+            "The Azure GPU SKU to use. See the ACI documentation for a list of "
+            "GPU SKUs available in each Azure region."
+        ),
+    )
+    memory: float = Field(
+        default=ACI_DEFAULT_MEMORY,
+        description=(
+            "The amount of memory in gigabytes to provide to the ACI task. Valid "
+            "amounts are specified in the Azure documentation. If not provided, a "
+            f"default value of  {ACI_DEFAULT_MEMORY} will be used unless present "
+            "on the task definition."
+        ),
+    )
+    subnet_ids: Optional[List[str]] = Field(
+        default=None,
+        description="A list of Azure subnet IDs the container should be connected to.",
+    )
+    stream_output: Optional[bool] = Field(
+        default=None,
+        description=(
+            "If `True`, logs will be streamed from the Prefect container to the local "
+            "console."
+        ),
+    )
+    env: Dict[str, Optional[str]] = Field(
+        title="Environment Variables",
+        default_factory=dict,
+        description=(
+            "Environment variables to provide to the task run. These variables are set "
+            "on the Prefect container at task runtime. These will not be set on the "
+            "task definition."
+        ),
+    )
+    # Execution settings
+    task_start_timeout_seconds: int = Field(
+        default=240,
+        description=(
+            "The amount of time to watch for the start of the ACI container. "
+            "before marking it as failed."
+        ),
+    )
+    task_watch_poll_interval: float = Field(
+        default=5.0,
+        description=(
+            "The amount of time to wait between Azure API calls while monitoring the "
+            "state of an Azure Container Instances task."
+        ),
+    )
+
+    @validator("command")
+    def validate_command(cls, command: List[str]):
+        """
+        A Pydantic validator that ensures a command is available to run
+        in an Azure container instance.
+
+        Args:
+            command: The command to validate.
+
+        Returns:
+            A command list if present, otherwise the base flow run command.
+        """
+
+        if not command:
+            return cls._base_flow_run_command()
```

Comment:
I don't think we should do this. This is populated by the `prepare_for_flow_run` method at runtime. This will freeze the command at block registration time.